### ISSUE TITLE
How to calculate RTM using deepwave

### ISSUE THREAD

**Issue by XuVV** [ar4/deepwave#58]
Hello, Doctor,
in the tutorial, it said that: "This example demonstrates inversion for the scattering potential with Born modelling, commonly known as imaging. We just run one iteration of optimisation, so this is essentially Reverse-Time Migration (RTM), but we could run more iteration to perform Least-Squares RTM (LSRTM)." 
but the result of RTM depends on the initial value of "scatter = torch.zeros_like(v_mig)". 
so how to calculate exact RTM using deepwave like devito?

**Comment by ar4** [comment]
Hello, and thank you for your question.

Unfortunately I am not sure what you mean by "exact RTM". The example
shows how to invert for the scattering model ("image") using the Born
propagator. In the example, as is typical for this task, we start with
a scattering model that is zero, and then use an optimizer to update
it to reduce the loss function value. The example only performs one
iteration of this update, which is known as RTM. If more iterations
were performed, this would be known as LSRTM. Can you tell me more
about what you want to do?

**Comment by XuVV** [comment]
Thanks for your reply. 

If I set the "scatter = torch.one_like(v_mig)", the result of one iteration is different from that of setting scatter = torch.one_like(v_mig), so which is what we want exact RTM result?

**Comment by XuVV** [comment]
Thank very much, Doctor. 

Yes, what I mean is regular RTM. I want to say that if I set `scatter = torch.one_like(v_mig)' insead of "scatter = torch.zeros_like(v_mig)", the results will be different.

Because I want to use deep wave to achieve standard LSRTM via classical optimization algorithm such as Conjugate gradient descent, so the gradient is needed. Could you please give some advice about it?

**Comment by ar4** [comment]
In RTM/LSRTM we are inverting for the perturbations from the background
velocity model. These perturbations are what I call the scattering model.
We should therefore start with our initial model being zero.

The RTM example performs one step of optimization - it calculates the
gradient and updates the scattering model (initially zero) using the
gradient. If you want to instead perform LSRTM you just have to run it for
more iterations. You can use the built-in optimizers in PyTorch, such as
the gradient descent optimizer used in the Deepwave RTM example, or
PyTorch's LBFGS optimizer. If you instead wish to write your own optimizer,
you can access the gradient using `scatter.grad.detach()` after running
backpropagation. You may wish to learn more about the basics of PyTorch
first, though, if you are unfamiliar with it.

**Comment by XuVV** [comment]
ok, thanks.

**Comment by ar4** [comment]
I am going to close this Issue, but please feel free to reopen it or to create a new one if you have any further questions.
